sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/BinaryOperationWithIdenticalExpressions.cs

```diff
@@ -40,6 +42,25 @@ public void Test(bool a, bool b)
             int i = 1 << 1;
             i = 1 << 0x1;
             i = 2 << 2; // Compliant
+
+            object.Equals(i, i);
+//                        ^ {{Correct one of the identical expressions on both sides of Object.Equals.}}
+//                           ^ Secondary@-1
+
+            var o = new object();
+
+            o.Equals(o);
+//          ^ {{Correct one of the identical expressions on both sides of Object.Equals.}}
+//                   ^ Secondary@-1
+
+            (new object()).Equals(new object());

```

Comment:
We've discussed that with @jccollet and decided to report on that.

There may be corner cases where this is a false positve (like Equals(list.Next, list.Next)), but that would clearly be a clear code smell.